builtin/providers/cloudstack/tags.go

```diff
@@ -0,0 +1,81 @@
+package cloudstack
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+// tagsSchema returns the schema to use for tags.
+//
+func tagsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeMap,
+		Optional: true,
+	}
+}
+
+// setTags is a helper to set the tags for a resource. It expects the
+// tags field to be named "tags"
+func setTags(cs *cloudstack.CloudStackClient, d *schema.ResourceData, resourcetype string) error {
+
+	if d.HasChange("tags") {
+		oraw, nraw := d.GetChange("tags")
+		o := oraw.(map[string]interface{})
+		n := nraw.(map[string]interface{})
+		create, remove := diffTags(tagsFromMap(o), tagsFromMap(n))
```

Comment:
I would expect that the `d.GetChange` func already returned the old and the new. So why not just delete all the old and create all the new? Even it there are overlapping values, it's still only one API call to remove all the old onesl and one API call to create all the new ones. I guess it would make the code a little cleaner...